### PR TITLE
Update parent pom and replace jupiter dependency

### PR DESCRIPTION
**Pull request by rjadhav-ch**
Updates pom to replace ch parent pom dependency. Adds a more specific `junit-jupiter-api` dependency in place of `junit-jupiter`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -43,7 +59,7 @@
 
 		<dependency>
 			<groupId>org.junit.jupiter</groupId>
-			<artifactId>junit-jupiter</artifactId>
+			<artifactId>junit-jupiter-api</artifactId>
```
Comment (VaughanJackson123) [inline pom.xml:62]: What is the motivation behind this change?

Comment (rjadhav-ch) [reply pom.xml:62]: It is needed for the annotation `@TestMethodOrder` and supports every other test requirements. `junit-jupiter` adds three other jars that aren't used in the project - `junit-jupiter`, `junit-jupiter-engine` and `junit-jupiter-params`.

Comment (VaughanJackson123) [reply pom.xml:62]: * I was thinking you must be right because clearly the PR had built and passed its tests. 
* I was confused because this seemed to go against this comment that recommended that just the `junit-jupiter` dependency be used for Jupiter >= 5.4: https://stackoverflow.com/questions/48448331/difference-between-junit-jupiter-api-and-junit-jupiter-engine (see `junit-jupiter artifact`).
* Given this contradiction, I took a look at the build: https://ci.platform.aws.chdev.org/teams/team-apollo/pipelines/item-handler/jobs/analyse-pull-request/builds/3:

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------

Results :

Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
```
